pennylane/ops/qubit.py

```diff
@@ -1191,6 +1263,43 @@ def _matrix(*params):
         return U
 
 
+class DiagonalQubitUnitary(DiagonalOperation):
+    r"""DiagonalQubitUnitary(D, wires)
+    Apply an arbitrary fixed diagonal unitary matrix.
+
+    **Details:**
+
+    * Number of wires: Any (the operation can act on any number of wires)
+    * Number of parameters: 1
+    * Gradient recipe: None
+
+    Args:
+        D (array[complex]): diagonal of unitary matrix
+        wires (Sequence[int] or int): the wire(s) the operation acts on
+    """
+    num_params = 1
+    num_wires = Any
+    par_domain = "A"
+    grad_method = None
+
+    @staticmethod
+    def _matrix(*params):
+        return np.diag(DiagonalQubitUnitary._diagonal(*params))
+
+    @staticmethod
+    def _diagonal(*params):
+        D = np.asarray(params[0])
+
+        if not np.allclose(D * D.conj(), np.ones_like(D)):
+            raise ValueError("Operator must be unitary.")
```

Comment:
Could this be a costly thing to check repetitively?

Reply:
I like the idea of having 'debug'/'production' modes for QNodes, where the user can turn off validations and checks when they have finished prototyping :) But this is a new PR/design decision.

Reply:
I essentially reproduced the code of `QubitUnitary` at this point. This is actually a problem more deeply rooted in how pennylane checks arguments. Currently, you can basically only define a type like "R" (reals) or "A" (array) and rudimentary checks are performed based on this. While this is fine for most gates it doesn't really fit the ones that have more complex parameters like `Hermitian`, `QubitUnitary` or `PauliRot`. 

It would basically be necessary to be able to override the parameter checks of the operations, but this should be ADRed first.